src/hoc/DragDropHandler.tsx

```diff
@@ -23,56 +22,49 @@ interface Prop {
 export const DragDropHandler = ({
   children,
   accept,
-  index,
   id,
   task,
-  moveCard,
   onDropItem,
   draggable = false,
   droppable = false, // New prop for droppable
 }: Prop) => {
   const { view } = useSelector(selectTaskBoard)
   const ref = useRef<HTMLDivElement | null>(null)
 
-  const [{ isOver, canDrop }, drop] = useDrop<{ task: TaskResponse }, unknown, { isOver: boolean; canDrop: boolean }>({
+  const [{ isOver, canDrop }, drop] = useDrop<{ id: string }, unknown, { isOver: boolean; canDrop: boolean }>({
     accept: accept,
     collect: (monitor: DropTargetMonitor) => ({
       isOver: monitor.isOver(),
       canDrop: monitor.canDrop(),
     }),
-    drop: (item, monitor) => {
+    drop: (item) => {
       if (onDropItem && id) {
         onDropItem({
-          taskId: item.task.id,
+          taskId: item.id,
           targetWorkflowStateId: id,
         })
       }
     },
     canDrop: () => droppable, // Only allow dropping if the component is droppable
   })
 
-  const [{ isDragging }, drag, preview] = useDrag({
-    type: accept,
-    item: () => {
-      return { task: task }
+  const [{ opacity }, drag, preview] = useDrag(
+    {
+      type: accept,
+      item: {
+        id: task?.id,
```

Comment:
i think we need to return the complete `task` object here because we need to show the `title` and `label` in the drag preview of list view task card.